client/unfoldable/unfoldable.js

```diff
@@ -0,0 +1,233 @@
+
+function convertJson(data) {
+
+  let id = typeof(data.root) == 'string' ? data.root : 'root'
+  let processed = []
+  let depth = 0
+
+  function replaceObjectIdsByActualObjects (id, data, processed, depth) {
+    let dataObjectId = id
+    let alreadyProcessed = processed.indexOf(id) != -1
```

Comment:
you can also use .includes() or ~processed.indexOf(id) to avoid the comparison. and if you are going to use comparison try to strive for deep comparison always.